lighthouse-core/test/lib/tracehouse/cpu-profile-model-test.js

```diff
@@ -167,31 +168,87 @@ describe('CPU Profiler Model', () => {
-        {startTime: ts(12.5e3), endTime: ts(13.4e3)},
+        {startTime: 12.5, endTime: 13.4},
         // A small task inside Foo, should move the end time of Bar to 15.7e3, start time of Baz to 16.8e3.
-        {startTime: ts(15.7e3), endTime: ts(16.8e3)},
+        {startTime: 15.7, endTime: 16.8},
       ];
 
       const events = CpuProfilerModel.synthesizeTraceEvents(profile, tasks);
 
       expect(events).toMatchObject([
         {ph: 'B', ts: ts(8.0e3), args: {data: {callFrame: {functionName: '(root)'}}}},
         {ph: 'B', ts: ts(8.0e3), args: {data: {callFrame: {functionName: '(program)'}}}},
-        {ph: 'B', ts: ts(8.0e3), args: {data: {callFrame: {functionName: 'Foo'}}}},
-        {ph: 'B', ts: ts(12.0e3), args: {data: {callFrame: {functionName: 'Bar'}}}},
+        {ph: 'B', ts: ts(9.0e3), args: {data: {callFrame: {functionName: 'Foo'}}}},
+        {ph: 'B', ts: ts(11.0e3), args: {data: {callFrame: {functionName: 'Bar'}}}},
         {ph: 'B', ts: ts(12.5e3), args: {data: {callFrame: {functionName: 'Baz'}}}},
         {ph: 'E', ts: ts(13.4e3), args: {data: {callFrame: {functionName: 'Baz'}}}},
         {ph: 'E', ts: ts(15.7e3), args: {data: {callFrame: {functionName: 'Bar'}}}},
         {ph: 'B', ts: ts(16.8e3), args: {data: {callFrame: {functionName: 'Baz'}}}},
-        {ph: 'E', ts: ts(17.5e3), args: {data: {callFrame: {functionName: 'Baz'}}}},
-        {ph: 'E', ts: ts(17.5e3), args: {data: {callFrame: {functionName: 'Foo'}}}},
-        {ph: 'E', ts: ts(19.0e3), args: {data: {callFrame: {functionName: '(program)'}}}},
-        {ph: 'E', ts: ts(19.0e3), args: {data: {callFrame: {functionName: '(root)'}}}},
+        {ph: 'E', ts: ts(17.4e3), args: {data: {callFrame: {functionName: 'Baz'}}}},
+        {ph: 'E', ts: ts(17.4e3), args: {data: {callFrame: {functionName: 'Foo'}}}},
+        {ph: 'E', ts: ts(19.5e3), args: {data: {callFrame: {functionName: '(program)'}}}},
+        {ph: 'E', ts: ts(19.5e3), args: {data: {callFrame: {functionName: '(root)'}}}},
+      ]);
+    });
+
+    it('should handle multiple task start/stop times with low sampling rate', () => {
+      /*
+        An artistic rendering of the below profile with tasks:
+        ████████████████(root)████████████████
+        ███████Task███████  ██████Task██████ █
+         ██████Eval██████   ██████Eval██████
+         ██████Foo███████   ██████Bar██████
+          █Fn█  █Fn█
+      */
+      profile = {
+        id: '0x1',
+        pid: 1,
+        tid: 1,
+        startTime: 9e6,
+        nodes: [
+          {id: 0, callFrame: {functionName: '(root)'}},
+          {id: 1, callFrame: {functionName: 'Foo', url: 'fileA.js'}, parent: 0},
+          {id: 2, callFrame: {functionName: 'Bar', url: 'fileA.js'}, parent: 0},
+        ],
+        samples: [0, 1, 1, 2, 2, 0],
+        timeDeltas: [0.5e3, 19.5e3, 20e3, 20e3, 20e3, 20e3],
+      };
+
+      const ts = x => profile.startTime + x;
+
+      // With the sampling profiler we know that Foo switched to Bar but not when.
```

Comment:
```suggestion
      // With the sampling profiler we know that Foo switched to Bar but not know when.
```

Reply:
Are you sure? :) I'm not sure how to read the new version.


Would you prefer...

```suggestion
      // With the sampling profiler we know that Foo switched to Bar, but we don't know when.
```

Reply:
Whoops, looks like my grammar was off on this one. Your second suggestion definitely sounds better though.